---
title: "lab5.Rmd"
author: "Minsoo"
date: "2/19/2020"
output: html_document
---

```{r message=FALSE}
library(tidyverse)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r eval=FALSE}
download.file(url="https://ndownloader.figshare.com/files/2292169",
              destfile = "data/portal_data_joined.csv")
```

Read in the data.
```{r}
surveys <- read_csv("data/portal_data_joined.csv")
```


Challenge 1. Using pipes, subset the surveys data to include animals collected before 1995 and retain only the columns year, sex, and weight.
```{r}
surveys_ysw <- surveys %>%
  filter(year < 1995) %>%
  select(year, sex, weight)
head(surveys_ysw, n=10)
```

Challenge 2.Create a new data frame from the surveys data that meets the following criteria: contains only the species_id column and a new column called hindfoot_half containing values that are half the hindfoot_length values. In this hindfoot_half column, there are no NAs and all values are less than 30.

Hint: think about how the commands should be ordered to produce this data frame!
```{r}
surveys %>% 
  mutate(hindfoot_half = hindfoot_length/2) %>% 
  filter(!is.na(hindfoot_half) & hindfoot_half < 30) %>% 
  select(species_id, hindfoot_half) %>% print(n=10)
```

Lab 5. Exercises
```{r}
SNPs <- read_tsv("data/23andMe_complete.txt", skip = 14, col_types = cols(chromosome = col_factor()))
```

Exercise 1. Subset (select) the 23andMe Complete table to just position and genotype
```{r}
select(SNPs, position, genotype) %>% print(n=10)
```

Exercise 2. Filter to just the MT chromosome and remove genotypes A and T.
```{r}
filter(SNPs, chromosome == "MT") %>% filter(genotype != "A" & genotype != "T") %>% print(n=10)
```

Exercise 3. Use group_by() and summarize() to find the min positon, max position and total number of positions for each chromosomes.
```{r}
group_by(SNPs, chromosome) %>% summarize(min_position= min(position), max_position = max(position), n=n()) %>% print(n=10)
```

Exercise 4.Building on ex3 create using mutate a new column with the density of SNPs for each chromosome based the max position - min position divide by the total number of SNPs
```{r}
group_by(SNPs, chromosome) %>% summarize(min_position= min(position), max_position = max(position), n=n()) %>% mutate(SNP_density = (max_position-min_position)/n) %>% print(n=10)
```

Exercise 5. Building on ex3 sort chromosomes based on SNP density.
```{r}
SNP_density <- group_by(SNPs, chromosome) %>% summarize(min_position= min(position), max_position = max(position), n=n()) %>% mutate(SNP_density = (max_position-min_position)/n) %>% arrange(SNP_density)
```

Exercise 6. Export the above table as a csv file to your data directory (There will be no results except the code to show me). However you will get an error when knitting if the code is not correct.
```{r}
write_csv(SNP_density, path = "data/SNP_density.csv")
```




